src/typechecker.erl

```diff
@@ -1139,6 +1139,18 @@ new_type_var() ->
     I = erlang:unique_integer(),
     "_TyVar" ++ integer_to_list(I).
 
+bounded_type_list_to_type(TEnv, Types) ->
+    case [ unfold_bounded_type(TEnv, Ty) || Ty <- Types ] of
+        []   -> type(none);
+        [Ty] -> Ty;
+        Tys  -> type(union, Tys)
```

Comment:
This is what the old `absform:function_type_list_to_fun_types/1` function did, but multiple specs really means intersection, not union.

Fixing this would require allowing intersection types (at least of functions) in the the type language. Currently they can only appear in the result of `expect_fun_type`. This is a non-trivial refactoring, so seeing how this PR doesn't make things worse, I'd be inclined to leave this for later.

Reply:
yeah, that union was added by me (from the assumption that the function should accept the union of argument types, which is not the same as union of fun types)

https://github.com/josefs/Gradualizer/commit/a1093d7afc1dd5b05b900b2d27a264a710c8c967#commitcomment-29533556

I am all for expanding the erl_parse:types() with intersection (and maybe others)

Reply:
Completely agree. Let's keep the union for now and create an issue about introducing the intersection type.

Reply:
I filed #97.